src/evo/specialtxman.cpp

```diff
@@ -151,6 +153,331 @@ bool CSpecialTxProcessor::CheckSpecialTx(const CTransaction& tx, const CBlockInd
+                newState->nVersion = opt_proTx->nVersion;
+                newState->pubKeyOperator = opt_proTx->pubKeyOperator;
+            }
+            newState->keyIDVoting = opt_proTx->keyIDVoting;
+            newState->scriptPayout = opt_proTx->scriptPayout;
+
+            newList.UpdateMN(opt_proTx->proTxHash, newState);
+
+            if (debugLogs) {
+                LogPrintf("CDeterministicMNManager::%s -- MN %s updated at height %d: %s\n",
+                    __func__, opt_proTx->proTxHash.ToString(), nHeight, opt_proTx->ToString());
+            }
+        } else if (tx.nType == TRANSACTION_PROVIDER_UPDATE_REVOKE) {
+            const auto opt_proTx = GetTxPayload<CProUpRevTx>(tx);
+            if (!opt_proTx) {
+                return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-protx-payload");
+            }
+
+            auto dmn = newList.GetMN(opt_proTx->proTxHash);
+            if (!dmn) {
+                return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-protx-hash");
+            }
+            auto newState = std::make_shared<CDeterministicMNState>(*dmn->pdmnState);
+            newState->ResetOperatorFields();
+            newState->BanIfNotBanned(nHeight);
+            newState->nRevocationReason = opt_proTx->nReason;
+
+            newList.UpdateMN(opt_proTx->proTxHash, newState);
+
+            if (debugLogs) {
+                LogPrintf("CDeterministicMNManager::%s -- MN %s revoked operator key at height %d: %s\n",
+                    __func__, opt_proTx->proTxHash.ToString(), nHeight, opt_proTx->ToString());
+            }
+        } else if (tx.nType == TRANSACTION_QUORUM_COMMITMENT) {
+            const auto opt_qc = GetTxPayload<llmq::CFinalCommitmentTxPayload>(tx);
+            if (!opt_qc) {
+                return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-qc-payload");
+            }
+            if (!opt_qc->commitment.IsNull()) {
+                const auto& llmq_params_opt = Params().GetLLMQ(opt_qc->commitment.llmqType);
+                if (!llmq_params_opt.has_value()) {
+                    return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-qc-commitment-type");
+                }
+                int qcnHeight = int(opt_qc->nHeight);
+                int quorumHeight = qcnHeight - (qcnHeight % llmq_params_opt->dkgInterval) + int(opt_qc->commitment.quorumIndex);
+                auto pQuorumBaseBlockIndex = pindexPrev->GetAncestor(quorumHeight);
+                if (!pQuorumBaseBlockIndex || pQuorumBaseBlockIndex->GetBlockHash() != opt_qc->commitment.quorumHash) {
+                    // we should actually never get into this case as validation should have caught it...but let's be sure
+                    return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS, "bad-qc-quorum-hash");
+                }
+
+                // The commitment has already been validated at this point, so it's safe to use members of it
```

Comment:
Please document that `GetAllQuorumMembers()` is now called by `BuildNewListFromBlock()` compared to current behavior in `develop`, otherwise changes are mostly move-only

`develop`:

https://github.com/dashpay/dash/blob/1083228079b06513fa43f97c3c693037347eb6c8/src/evo/deterministicmns.cpp#L996-L1003

PR:

https://github.com/dashpay/dash/blob/8546f331702c4284692a9b793b85a1a27a206deb/src/evo/specialtxman.cpp#L410-L413

Reply:
I updated PR description